dsalgo/search.py

```diff
@@ -0,0 +1,146 @@
+import sys
+import math
+
+class Search:
+
+    def __new__(self, array, element, algo):
+        '''
+        args:
+            :array: (list) : a python list
+            :algo: (str): searching algorithm type
+                            values supported are:
+                            1. Linear Search (linaer)
+                            2. Binary Search (binary)
+                            3. Jump Search (jump)
+                            4. Fibonacci Search (fibonacci)
+            :element -> element that want to be search
+        return:
+            index of search element
+        '''
+        self.array = array
+        self.algo = algo
+        self.element = element
+
+        if self.algo == 'linear':
+            return LinearSearch(self.array, self.element)
+        elif self.algo == 'binary':
+            return BinarySearch(self.array, self.element)
+        elif self.algo == 'jump':
+            return JumpSearch(self.array, self.element)
+        elif self.algo == 'fibonacci':
+            return FibonacciSearch(self.array, self.element)
+        else:
+            sys.stderr.write("Error: unsupported searchning algorithm passed!")
+
+
+def LinearSearch(array, element):
+    '''
+    In linear search, a sequential search is made over all 
+    items one by one. Every item is checked and if a match 
+    is found then that particular item is returned, 
+    otherwise the search continues till the end of the data collection.
+    
+    args:
+        :array:(list) -> list to be sorted
+        :element -> element that want to be search
+    '''
+    for i in range(len(array)):
+        if array[i] == element:
+            return i
+    return -1
+
+
+def BinarySearch(array, element):
+    '''
+    Binary search follows a divide and conquer methodology. 
+    It is faster than linear search but requires that 
+    the array be sorted before the algorithm is executed.
+    
+    If array is not sorted then it will give and worng answer to you.
+    
+    args:
+        :array:(list) -> list to be sorted
+        :element -> element that want to be search
+    '''
+    first = 0
```

Comment:
add a conditional check to check whether the passed array is sorted or not to avoid unexpected results.